repository/src/main/java/org/apache/atlas/repository/graph/IndexRecoveryService.java

```diff
@@ -63,14 +65,16 @@ public class IndexRecoveryService implements Service, ActiveStateChangeHandler {
     private       RecoveryThread         recoveryThread;
 
     @Inject
-    public IndexRecoveryService(Configuration config, AtlasGraph graph) {
+    public IndexRecoveryService(Configuration config, AtlasGraph graph, ICuratorFactory curatorFactory) {
         this.configuration               = config;
         this.isIndexRecoveryEnabled      = config.getBoolean(ApplicationProperties.INDEX_RECOVERY_CONF, DEFAULT_INDEX_RECOVERY);
         long recoveryStartTimeFromConfig = getRecoveryStartTimeFromConfig(config);
         long healthCheckFrequencyMillis  = config.getLong(SOLR_STATUS_CHECK_RETRY_INTERVAL, SOLR_STATUS_RETRY_DEFAULT_MS);
         this.recoveryInfoManagement      = new RecoveryInfoManagement(graph);
 
-        this.recoveryThread = new RecoveryThread(recoveryInfoManagement, graph, recoveryStartTimeFromConfig, healthCheckFrequencyMillis);
+        final InterProcessMutex indexRecoveryLock = curatorFactory.lockInstanceWithDefaultZkRoot(INDEX_RECOVERY_LOCK);
```

Comment:
Here we are calling this method without check activeactiveenable
In tasks, we are calling this just before acquiring lock and only if activeactiveenabled
and in typedef we are using a slightly diff method.

Reply:
Please review this and let's discuss if possible

Reply:
Addressed this by making all the places consistent